New test failures when Icechunk bumped to `0.2.0`

Updating the min version of Icechunk to `0.2.0` caused [four test failures in the CI](https://github.com/zarr-developers/VirtualiZarr/actions/runs/13446771371/job/37573706608?pr=443) related to appending to Icechunk stores. 

```python

FAILED virtualizarr/tests/test_writers/test_icechunk.py::TestAppend::test_append_virtual_ref_without_encoding - KeyError: '///foo'
FAILED virtualizarr/tests/test_writers/test_icechunk.py::TestAppend::test_append_virtual_ref_with_encoding - KeyError: '///air'
FAILED virtualizarr/tests/test_writers/test_icechunk.py::TestAppend::test_append_with_compression_succeeds[2] - KeyError: '///air'
FAILED virtualizarr/tests/test_writers/test_icechunk.py::TestAppend::test_append_with_compression_succeeds[3] - KeyError: '///air'
```

related: https://github.com/zarr-developers/VirtualiZarr/pull/443#issuecomment-2672957610

--- a/ci/environment.yml
+++ b/ci/environment.yml
@@ -15,7 +15,7 @@ dependencies:
   - universal_pathlib
   - hdf5plugin
   - numcodecs
-  - icechunk>=0.1.1
+  - icechunk>=0.2.0
   # Testing
   - codecov[toml]
   - pre-commit
@@ -37,4 +37,4 @@ dependencies:
   - astropy
   - pip
   - pip:
-    - imagecodecs-numcodecs==2024.6.1
+      - imagecodecs-numcodecs==2024.6.1

--- a/ci/upstream.yml
+++ b/ci/upstream.yml
@@ -28,7 +28,7 @@ dependencies:
   - pooch
   - fsspec
   - pip
+  - icechunk>=0.2.0
   - pip:
-      - git+https://github.com/earth-mover/icechunk.git@main#subdirectory=icechunk-python # Installs zarr-python v3.0.0 as dependency
       - git+https://github.com/fsspec/kerchunk.git@main
       - imagecodecs-numcodecs==2024.6.1

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -71,7 +71,7 @@ all_readers = [
 
 # writers
 icechunk = [
-    "icechunk>=0.1.1",
+    "icechunk>=0.2.0",
 ]
 kerchunk = ["fastparquet"]
 all_writers = [

--- a/virtualizarr/writers/icechunk.py
+++ b/virtualizarr/writers/icechunk.py
@@ -305,7 +305,7 @@ def write_manifest_virtual_refs(
     if group.name == "/":
         key_prefix = arr_name
     else:
-        key_prefix = f"{group.name}{arr_name}"
+        key_prefix = f"{group.name}/{arr_name}"
 
     # loop over every reference in the ChunkManifest for that array
     # TODO inefficient: this should be replaced with something that sets all (new) references for the array at once

--- a/virtualizarr/writers/icechunk.py
+++ b/virtualizarr/writers/icechunk.py
@@ -302,7 +302,6 @@ def write_manifest_virtual_refs(
 ) -> None:
     """Write all the virtual references for one array manifest at once."""
 
-    # perhaps something changed that group.name now defaults to "/" instead of ""?
     if group.name == "/":
         key_prefix = arr_name
     else:

--- a/virtualizarr/writers/icechunk.py
+++ b/virtualizarr/writers/icechunk.py
@@ -302,7 +302,11 @@ def write_manifest_virtual_refs(
 ) -> None:
     """Write all the virtual references for one array manifest at once."""
 
-    key_prefix = f"{group.name}/{arr_name}"
+    # perhaps something changed that group.name now defaults to "/" instead of ""?
+    if group.name == "/":
+        key_prefix = arr_name
+    else:
+        key_prefix = f"{group.name}{arr_name}"
 
     # loop over every reference in the ChunkManifest for that array
     # TODO inefficient: this should be replaced with something that sets all (new) references for the array at once
@@ -324,8 +328,11 @@ def write_manifest_virtual_refs(
         chunk_key = generate_chunk_key(index, append_axis, existing_num_chunks)
 
         # set each reference individually
+        import pdb
+
+        pdb.set_trace()
         store.set_virtual_ref(
-            # TODO it would be marginally neater if I could pass the group and name as separate args
+            # perhaps an additional "/" is added here as well?
             key=f"{key_prefix}/c/{chunk_key}",  # should be of form 'group/arr_name/c/0/1/2', where c stands for chunks
             location=path.item(),
             offset=offset.item(),

--- a/virtualizarr/writers/icechunk.py
+++ b/virtualizarr/writers/icechunk.py
@@ -328,11 +328,8 @@ def write_manifest_virtual_refs(
         chunk_key = generate_chunk_key(index, append_axis, existing_num_chunks)
 
         # set each reference individually
-        import pdb
-
-        pdb.set_trace()
         store.set_virtual_ref(
-            # perhaps an additional "/" is added here as well?
+            # TODO it would be marginally neater if I could pass the group and name as separate args
             key=f"{key_prefix}/c/{chunk_key}",  # should be of form 'group/arr_name/c/0/1/2', where c stands for chunks
             location=path.item(),
             offset=offset.item(),

